### PR TITLE
Decoder Performance optimization and Intrinsic fixes

### PR DESCRIPTION
**Pull request by nijilk**
1. Performance improvement in decoder mainly by removing lot of control codes.
2. Fixed few intrinsic functions which were causing memory corruption.

### CODE REVIEW

**Comment by nijilk** [comment]
@joelsole @hassount 
Please review

File: Source/Lib/Common/ASM_SSE4_1/EbWarpPlane_highbd_SSE4_1.c [hunk 1]
```diff
@@ -0,0 +1,646 @@
...
+                        ->dst[(i + k + 4) * conv_params->dst_stride + j];
+                    res_lo = _mm_add_epi32(res_lo, res_add_const);
+                    res_lo = _mm_sra_epi32(_mm_add_epi32(res_lo, reduce_bits_vert_const),
+                        reduce_bits_vert_shift);
+
+                    if (conv_params->do_average) {
+                        __m128i *const dst16 = (__m128i *)&pred[(i + k + 4) * p_stride + j];
+                        __m128i p_32 = _mm_cvtepu16_epi32(_mm_loadl_epi64(p));
+
+                        if (conv_params->use_dist_wtd_comp_avg) {
```
Comment (spawlows) [inline Source/Lib/Common/ASM_SSE4_1/EbWarpPlane_highbd_SSE4_1.c]: It should be "if (conv_params->use_jnt_comp_avg)" in all places to match C implementation.  PR #846 have that same change (eb_av1_highbd_warp_affine_sse4_1), with working UT.

Comment (nijilk) [reply Source/Lib/Common/ASM_SSE4_1/EbWarpPlane_highbd_SSE4_1.c]: > 
> 
> It should be "if (conv_params->use_jnt_comp_avg)" in all places to match C implementation. PR #846 have that same change (eb_av1_highbd_warp_affine_sse4_1), with working UT.

@spawlows Issue is fixed

Comment (spawlows) [reply Source/Lib/Common/ASM_SSE4_1/EbWarpPlane_highbd_SSE4_1.c]: @nijilk Please rebase, and remove from this commit analogous changes which are merged to master from PR #846 (eb_av1_highbd_warp_affine_sse4_1)

Comment (nijilk) [reply Source/Lib/Common/ASM_SSE4_1/EbWarpPlane_highbd_SSE4_1.c]: @spawlows I have already rebased to master after removing the analogous changes.
Please review.

**Comment by hassount** [comment]
@joelsole is this good to go?

**Comment by joelsole** [comment]
yes, lgtm 
There is a trivial conflict in EbDefinitions.h

**Comment by hassount** [comment]
Great, @nijilk can you please rebase, solve the conflicts and squash the commits to as the smallest possible number of commits

**Comment by nijilk** [comment]
> 
> 
> Great, @nijilk can you please rebase, solve the conflicts and squash the commits to as the smallest possible number of commits

@hassount Re-based and Squashed the changes. Please review.

**Pull request merged**